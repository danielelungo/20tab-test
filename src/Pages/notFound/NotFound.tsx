import { FC } from "react"
import { Link } from "react-router-dom"

const NotFound: FC = () => {
  return (
    <div>
      <h1>404: Page Non Found</h1>
      <Link to={"/"}>Go Back To Home Page</Link>
    </div>
  )
}

export default NotFound
