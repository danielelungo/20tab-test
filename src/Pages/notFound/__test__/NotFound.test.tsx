import { render, screen } from "@testing-library/react"
import { BrowserRouter } from "react-router-dom"
import renderer from "react-test-renderer"
import NotFound from "../NotFound"

const MockNotFound = () => {
  return (
    <BrowserRouter>
      <NotFound />
    </BrowserRouter>
  )
}

describe("NotFound component", () => {
  test("rmatch NotFound SnapShot", () => {
    const tree = renderer.create(<MockNotFound />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  test("Should render header", () => {
    render(<MockNotFound />)
    const headingElement = screen.getByRole("heading", { name: "404: Page Non Found" })
    expect(headingElement).toBeInTheDocument()
  })
})
