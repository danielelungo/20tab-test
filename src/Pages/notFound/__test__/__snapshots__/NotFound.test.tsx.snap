// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound component rmatch NotFound SnapShot 1`] = `
<div>
  <h1>
    404: Page Non Found
  </h1>
  <a
    href="/"
    onClick={[Function]}
  >
    Go Back To Home Page
  </a>
</div>
`;
